internal/parser_test.go

```diff
@@ -126,3 +129,100 @@ func findTargetNode(doc *Node) *Node {
 	})
 	return target
 }
+
+func fixturesParseFragmentWithOptions() []struct {
+	name   string
+	source string
+	want   []*Node
+	panic  string
+} {
+	return []struct {
+		name   string
+		source string
+		want   []*Node
+		panic  string
+	}{
+		{
+			name:   "none",
+			source: "<div />",
+			want:   []*Node{{Type: ElementNode, DataAtom: atom.Div, Data: atom.Div.String(), Namespace: "", Loc: []loc.Loc{{Start: 1}}}},
+		},
+		{
+			name:   "fault input currently accepted",
+			source: `<A { 0>`,
+			want:   []*Node{{Component: true, Type: ElementNode, DataAtom: 0, Data: "A", Namespace: "", Attr: []Attribute{{Namespace: "", Key: " 0>", KeyLoc: loc.Loc{Start: 4}, Val: "", ValLoc: loc.Loc{Start: 7}, Tokenizer: nil, Type: ShorthandAttribute}}, Loc: []loc.Loc{{Start: 1}}}},
```

Comment:
@natemoo-re determined that this is a faulty parse / tokenization, this should be resolved probably as an element w/ two empty attributes with names `{` and `0`?? like `<A {={true} 0={true}>`?

to give context to the failing test

Reply:
hmm but from the test fail we can see that it gets parsed as `<A {0>}/>` as if `0>` was a valid identifier in JS, resulting in the printed output of `<A 0>={0>}/>` similar to `<A {x}/>` == `<A x={x}/>`

Reply:
I can actually make this test pass by altering

https://github.com/withastro/compiler/blob/2dc9f533b0afbdb7a17d7e5a283b6545b7e53feb/internal/print-to-source.go#L60-L63

since currently we don't strip whitespace from the attribute name but we do from the attribute variable?